cmake/EthOptions.cmake

```diff
@@ -11,14 +11,59 @@ macro(configure_project)
 	option(VMTRACE "Enable VM tracing" OFF)
 	option(PROFILING "Enable profiling (deprecated)" OFF)
 	option(FATDB "Enable fat state database" ON)
-	option(ROCKSDB "Build with RocksDB instead of LevelDB" OFF)
 	option(PARANOID "Enable additional checks when validating transactions (deprecated)" OFF)
 	option(MINIUPNPC "Build with UPnP support" OFF)
 	option(FASTCTEST "Enable fast ctest" OFF)
 
+	set(supported_dbs LEVELDB ROCKSDB LMDB)
+	string(REPLACE ";" ", " supported_dbs_str "${supported_dbs}")
+	set(DB LEVELDB CACHE STRING "Database implementation to use (${supported_dbs_str})")
+	set_property(CACHE DB PROPERTY STRINGS ${supported_dbs})
+	set(valid_db OFF)
+	foreach(supported_db ${supported_dbs})
+		if(DB STREQUAL supported_db)
+			set(valid_db ON)
+			break()
+		endif()
+	endforeach()
+	if(NOT valid_db)
+		message(WARNING
+			"Invalid database selected ${DB}, must be one of"
+			" ${supported_dbs_str}, using LEVELDB instead")
+		set(DB LEVELDB CACHE STRING
+			  "Database implementation to use (${supported_dbs_str})"
+				FORCE)
+	endif()
+
+	add_definitions("-DETH_${DB}")
+	if(HUNTER_ENABLED)
+		if(DB STREQUAL "LMDB")
+			set(hunter_name lmdbxx)
+		else()
+			string(TOLOWER "${DB}" hunter_name)
+		endif()
+		hunter_add_package(${hunter_name})
+
+		if(DB STREQUAL LMDB)
+			find_package(lmdbxx CONFIG REQUIRED)
+		elseif(DB STREQUAL ROCKSDB)
+			find_package(RocksDB CONFIG REQUIRED)
+		else()
+			find_package(leveldb CONFIG REQUIRED)
```

Comment:
Do not take leveldb from Hunter until it supports Windows. So: lmdbxx and rocksdb from Hunter only, leveldb from system only.

Reply:
OK I will update other diff with the advice from here. Still not sure how any version of leveldb is working on Windows unless it is the forked version from bitcoin.